org.eclipse.jdt.core/model/org/eclipse/jdt/internal/compiler/ICompilerFactory.java

```diff
@@ -28,11 +29,15 @@ public interface ICompilerFactory {
 	 * @param compilerConfig - the configuration to control the compiler behavior
 	 * @param requestor - the requestor to receive and persist compilation results
 	 * @param problemFactory - the factory to create problem descriptors
+	 * @param compilationProgress - the CompilationProgress to be used
 	 * @return the new compiler instance
+	 * @see CompilationProgress
 	 */
-	public Compiler newCompiler(final INameEnvironment environment,
-			final IErrorHandlingPolicy policy,
-			final CompilerConfiguration compilerConfig,
-			final ICompilerRequestor requestor,
-			final IProblemFactory problemFactory);
+	public Compiler newCompiler(INameEnvironment environment,
```

Comment:
Would it be possible for a 1st iteration to add the new construtor and keep the other as deprecated, and plan removal for ~4 days later?
We should be able to update promptly, but we'd rather avoid broken builds if possible.

Reply:
The workflow would be ok for me, but it feels like too much effort to actually call this or that constructor depending on availability. So if you insist on it would it be possible that you add that extra code?

Reply:
OK, I understand the trouble and agree it's probably not worth it. I'm fine merging it at your own convenience; we'll easily spot the issue in the javac-based branch afterwards, and even if build is broken for a couple of days, that wouldn't be a strong issue; so let's not make it a cause of a delay upstream.

Reply:
would you prefer a merge asap or on m1?

Reply:
ASAP would work fine.